packages/jest-resolve/src/__tests__/resolve.test.js

```diff
@@ -181,3 +181,32 @@ describe('getMockModule', () => {
     );
   });
 });
+
+describe('Resolver.getModulePaths() -> nodeModulesPaths()', () => {
+  let moduleMap;
+  beforeEach(() => {
+    moduleMap = new ModuleMap({
+      duplicates: [],
+      map: [],
+      mocks: [],
+    });
+  });
+
+  it('can resolve node modules relative to absolute paths in "moduleDirectories"', () => {
+    const cwd = 'D:\\project';
```

Comment:
this won't pass on a mac or linux. Can you switch on `os.platform === 'win32'` and use unix style paths? We run CI on appveyor, so it should still guard against regression